readme/doc_resource.go

```diff
@@ -527,15 +588,183 @@ func (r *docResource) Delete(
 	}
 
 	requestOpts := apiRequestOptions(state.Version)
-	tflog.Info(ctx, fmt.Sprintf("deleting doc with request options=%+v", requestOpts))
 
-	// Delete the doc.
-	_, apiResponse, err := r.client.Doc.Delete(state.Slug.ValueString(), requestOpts)
+	// Check the category's docs to find the doc and its children.
+	docs, _, err := r.client.Category.GetDocs(state.CategorySlug.ValueString(), requestOpts)
 	if err != nil {
-		resp.Diagnostics.AddError("Unable to delete doc", clientError(err, apiResponse))
+		resp.Diagnostics.AddError("Unable to retrieve category docs.", clientError(err, nil))
+		return
+	}
+
+	// Gather the slugs of the docs to delete, including the parent and its children.
+	slugsToDelete, parentDoc := r.identifyDocsToDelete(ctx, state.Slug.ValueString(), docs, resp)
+	if parentDoc == nil {
+		return
+	}
+
+	// Ensure children are handled correctly.
+	// Delete them first if DestroyChildDocs is enabled. Otherwise, return.
+	if !r.handleChildDocs(ctx, resp, state.Slug.ValueString(), parentDoc) {
+		return
+	}
+
+	// Perform deletions
+	r.deleteDocs(ctx, resp, slugsToDelete, requestOpts)
+
+	// Warn if child docs were deleted
+	if len(slugsToDelete) > 1 && r.config.DestroyChildDocs.ValueBool() {
+		resp.Diagnostics.AddWarning(
+			fmt.Sprintf("Doc '%s' and its child docs were destroyed!", state.Slug.ValueString()),
+			fmt.Sprintf("The provider configuration 'config.destroy_child_docs' is set to true. "+
+				"Child docs were deleted before the parent doc '%s': %s", state.Slug.ValueString(),
+				strings.Join(slugsToDelete[1:], ", ")),
+		)
```

Comment:
This section is the fix.

It refactors things to properly iterate over child docs and discover parent/child relationships to delete docs in the proper order.